### ISSUE TITLE
memory leak in glances container

### ISSUE THREAD

**Issue by kkellner** [hassio-addons/addon-glances#61]
# Problem/Motivation

When running glances add-on with influxdb configured, the glances container continous to grow in memory consumption.   280MB to 600MB in two days.

Memory consumption. Note the sharp drops were hass (hassio) restarts.
![image](https://user-images.githubusercontent.com/9540241/78516586-5343fe80-7777-11ea-9215-c3a593baf6f4.png)

From within container:
![image](https://user-images.githubusercontent.com/9540241/78516751-eda44200-7777-11ea-9ecf-9f33d5262d27.png)

## Expected behavior

Memory hits a stable state.

## Actual behavior

Memory consumption continues to climb. 

## Steps to reproduce

Using HA v0.107.7, configure glances plugin v0.7.1 as follows and wait a day or two to watch memory climb.

```yaml
log_level: info
process_info: false
refresh_time: 10
ssl: false
certfile: fullchain.pem
keyfile: privkey.pem
influxdb:
  enabled: true
  host: a0d7b954-influxdb
  port: 8086
  username: glances
  password: mypasword
  database: glances
  prefix: localhost
  interval: 60
```

## Proposed changes

none

**Comment by kkellner** [comment]
This issue may be related to this:  https://github.com/nicolargo/glances/issues/1341

I changed value of `history_size` in file `/config/glances/glances.conf` from:
```
history_size=28800
```
to
```
history_size=0
```
I bounced glances container.   It will take a day or two to see if memory climbs.

**Comment by kkellner** [comment]
So far so good with `history_size=0` change.....   glances container memory for past 16 hours has been flat with not noticable change in CPU.

Change was implemented at ~1am on the following chart:

![image](https://user-images.githubusercontent.com/9540241/78616385-4b489500-7831-11ea-9988-5b2625929797.png)

**Comment by SebLz** [comment]
Hello, exact same behavior observed on my system (ha 0.108.6, glances 0.7.1, arm 64 hardware (beelink gt1 - 2GB ram)). I have an automation script restarting glance container when swap usage > 90%. This happens roughly every 1 or 2 days.

`- id: purge_swap_when_full
  alias: Purge SWAP file when almost full
  description: ''
  trigger:
  - entity_id: sensor.glances_swap_used_percent
    platform: numeric_state
    above: 90
    for:
      minutes: 10
  condition: []
  action:
  - service: persistent_notification.create
    data:
      message: 'High SWAP usage: {{ states(''sensor.glances_swap_used_percent'') }}, restarting glances addon'
      title: High SWAP usage
  - service: hassio.addon_restart
    data:
      addon: a0d7b954_glances`

**Comment by sinclairpaul** [comment]
Although I've just added a PR for this, tbh I'm confused as the previous comments are that /config/glances/glances.conf was edited.

As you can see at https://github.com/hassio-addons/addon-glances/blob/dbe6d21d70e24cc0794dc2a0cd83191bcaefa927/glances/rootfs/etc/services.d/glances/run#L10

it uses the /etc/glances.conf file, unless something is broken here??

**Comment by kkellner** [comment]
Now sure what config glances is ultimately using when its runing.   I know when I edited `/config/glances/glances.conf`, the `history_size=0` value I set there was used at runtime.

**Comment by sinclairpaul** [comment]
Was me being dumb, the file is copied from config to /etc on startup.

As you have access to the config and can edit it, is there a reason that this issue should remain open?  It also appears to be an upstream issue.

**Comment by kkellner** [comment]
Is the default now to set `history_size=0` in `glances.conf`?  If so, this issue can be closed.

**Comment by sinclairpaul** [comment]
Config matches upstream default settings. @frenck I will let you decide if this should be closed or not.

**Comment by frenck** [comment]
Addressed in 302a49a
